homework-classes/App.js

```diff
@@ -1,78 +1,100 @@
-    this.mainContainer = null;
-    this.initialize(url);
+    this.createHeader(url);
+  }
+
+  createHeader(url) {
+    const header = Util.createAndAppend('header', root, {
+      class: 'container jumbotron mt-3 mb-3 px-5',
+    });
+    Util.createAndAppend('main', root, {
+      id: 'container',
+      class: 'container align-items-start',
+    });
+    const headerDiv = Util.createAndAppend('div', header, {
+      class: 'header-group d-flex flex-row justify-content-between mb-5 px-5',
+    });
+    Util.createAndAppend('h2', headerDiv, {
+      text: 'Hack Your Future Repositories',
+      class: 'display-4 d-flex align-items-center mx-auto col-sm header-title',
+    });
+    const hyfLink = Util.createAndAppend('a', headerDiv, {
+      href: 'https://www.hackyourfuture.net/',
+      target: '_blank',
+    });
+    Util.createAndAppend('img', hyfLink, {
+      class: 'img-fluid rounded img-thumbnail mx-auto',
+      src: './hyf.png',
+      alt: 'hack your future thumbnail',
+    });
+    Util.createAndAppend('select', header, {
+      id: 'repo-select',
+      class: 'form-control',
+      'aria-label': 'Hack Your Future Repositories Selection',
+    });
+    this.fetchRepositories(url);
   }
 
   /**
-   * Initialization
    * @param {string} url The GitHub URL for obtaining the organization's repositories.
    */
-  async initialize(url) {
-    // Add code here to initialize your app
-    // 1. Create the fixed HTML elements of your page
-    // 2. Make an initial XMLHttpRequest using Util.fetchJSON() to populate your <select> element
-
-    const root = document.getElementById('root');
-    const header = Util.createAndAppend('header', root, { class: 'header' });
-    this.mainContainer = Util.createAndAppend('div', root, { id: 'container' });
 
+  async fetchRepositories(url) {
     try {
       const repos = await Util.fetchJSON(url);
       this.repos = repos.map(repo => new Repository(repo));
-      // TODO: add your own code here
+      this.appendRepositoriesToSelect(repos);
     } catch (error) {
       this.renderError(error);
     }
   }
 
-  /**
-   * Removes all child elements from a container element
-   * @param {*} container Container element to clear
-   */
-  clearContainer() {
-    while (this.mainContainer.firstChild) {
-      this.mainContainer.removeChild(this.mainContainer.firstChild);
-    }
+  appendRepositoriesToSelect(repos) {
+    const selectList = document.getElementById('repo-select');
+    repos
+      .sort((a, b) => a.name.localeCompare(b.name))
+      .forEach((repository, index) => {
+        Util.createAndAppend('option', selectList, {
+          text: repository.name,
+          value: index,
+        });
+      });
+    this.defaultRepo = new Repository(repos[0]).render();
+    this.fetchContributor(repos[0].contributors_url);
+    selectList.addEventListener('change', () => {
+      root.lastChild.innerHTML = '';
+      this.selectedRepo = new Repository(repos[selectList.value]).render();
```

Comment:
Do not create a new Repository object here. Reuse what you already created earlier.